src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,417 @@
+            StringBuilder skipTokenBuilder = new StringBuilder(String.Empty);
+            if (propertiesForSkipToken == null)
+            {
+                return skipTokenBuilder.ToString();
+            }
+
+            int count = 0;
+            string uriLiteral;
+            object value;
+            int lastIndex = propertiesForSkipToken.Count() - 1;
+
+            foreach (IEdmProperty property in propertiesForSkipToken)
+            {
+                bool islast = count == lastIndex;
+                string propertyName = EdmLibHelpers.GetClrPropertyName(property, model);
+                IEdmStructuredObject obj = lastMember as IEdmStructuredObject;
+                if (obj != null)
+                {
+                    obj.TryGetPropertyValue(propertyName, out value);
+                }
+                else
+                {
+                    value = lastMember.GetType().GetProperty(propertyName).GetValue(lastMember);
+                }
+
+                if (value == null)
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401);
+                }
+                else if (property.Type.IsEnum())
+                {
+                    ODataEnumValue enumValue = new ODataEnumValue(value.ToString(), value.GetType().FullName);
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(enumValue, ODataVersion.V401, model);
+                }
+                else
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401, model);
+                }
+
+                skipTokenBuilder.Append(propertyName).Append(propertyDelimiter).Append(uriLiteral).Append(islast ? String.Empty : CommaDelimiter.ToString());
+                count++;
+            }
+
+            return skipTokenBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Apply the $skiptoken query to the given IQueryable.
+        /// </summary>
+        /// <param name="query">The original <see cref="IQueryable"/>.</param>
+        /// <param name="skipTokenQueryOption">The skiptoken query option which needs to be applied to this query option.</param>
+        /// <returns>The new <see cref="IQueryable"/> after the skiptoken query has been applied to.</returns>
+        public override IQueryable<T> ApplyTo<T>(IQueryable<T> query, SkipTokenQueryOption skipTokenQueryOption)
+        {
+            if (skipTokenQueryOption == null)
+            {
+                throw Error.ArgumentNullOrEmpty("skipTokenQueryOption");
+            }
+
+            ODataQuerySettings querySettings = skipTokenQueryOption.QuerySettings;
+            IList<OrderByNode> orderByNodes = skipTokenQueryOption.OrderByNodes;
+            return ApplyToCore(query, querySettings, orderByNodes, skipTokenQueryOption.Context, skipTokenQueryOption.RawValue) as IOrderedQueryable<T>;
+        }
```

Comment:
Should this just call 
    return ApplyTo(query, skipTokenQueryOption) as IOrderedQueryable<T>
?